services/src/main/java/org/exoplatform/addons/gamification/search/RuleSearchConnector.java

```diff
@@ -60,7 +60,9 @@ public class RuleSearchConnector {
       + "            }\n"
       + "          }\n"
       + "        }\n";
-
+  
+  private static final String          ILLEGAL_SEARCH_CHARACTERS    = "\\!?^()+-=<>{}[]:\"'*~&|";
```

Comment:
The characters to escape with `\` are listed here (avoid adding others please)
https://www.elastic.co/guide/en/elasticsearch/reference/7.17/query-dsl-simple-query-string-query.html#simple-query-string-query-notes

Reply:
those are the reserved characters for simple query string 
but we need to escape the reserved characters for query string :
 https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-query-string-query.html#_reserved_characters